streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -550,6 +561,10 @@ void runLoop() {
         // until the rebalance is completed before we close and commit the tasks
         while (isRunning() || taskManager.isRebalanceInProgress()) {
             try {
+                if (shutdownRequested.get()) {
+                    sendShutdownRequest(shutdownTypeRequested);
+                    return;
```

Comment:
if we are shutting down may as well start now instead of waiting for the rebalance. this will stop every thread

Reply:
I can't tell; are we guaranteed to actually send the joinGroup request by now?

Maybe it's safer to just keep running this loop until we get back the "you should shut down" response assignment. We _could/should_ add a condition into `runOnce` so that we don't actually process anything once we have `shutdownRequested` set.

Reply:
you are right again. See previous reponse